pylearn2/datasets/svhn.py

```diff
@@ -214,11 +233,11 @@ def split_train_valid(path, num_valid_train=400,
             train_index = list(train_index)
 
             train_x = numpy.concatenate((train_x,
-                                         data['X'][:, :, :, train_index]),
+                                        data['X'][:, :, :, train_index]),
                                         axis=3)
             train_y = numpy.concatenate((train_y, data['y'][train_index, :]))
             valid_x = numpy.concatenate((valid_x,
-                                         data['X'][:, :, :, valid_index]),
+                                        data['X'][:, :, :, valid_index]),
```

Comment:
Here is the story about one_hot. It did not support it originally, but someone added the functionality without testing and that's why it's broken. There is a separate PR #1371 to fix the issue